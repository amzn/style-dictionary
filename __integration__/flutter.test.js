/*
 * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

const fs = require('fs-extra');
const StyleDictionary = require('../index');
const {buildPath} = require('./_constants');

describe('integration', () => {
  describe('flutter', () => {
    StyleDictionary.extend({
      source: [`__integration__/tokens/**/*.json?(c)`],
      platforms: {
        flutter: {
          transformGroup: `flutter`,
          buildPath,
          files: [{
            destination: "style_dictionary.dart",
            format: "flutter/class.dart",
            className: "StyleDictionary"
          },{
            destination: "style_dictionary_with_references.dart",
            format: "flutter/class.dart",
            className: "StyleDictionary",
            options: {
              outputReferences: true
            }
          }]
        },
        flutter_separate: {
          transformGroup: `flutter-separate`,
          buildPath,
          files: [{
            destination: "style_dictionary_color.dart",
            format: "flutter/class.dart",
            className: "StyleDictionaryColor",
            type: "color",
            filter: {
              attributes: {
                category: "color"
              }
            }
          },{
            destination: "style_dictionary_sizes.dart",
            format: "flutter/class.dart",
            className: "StyleDictionarySize",
            type: "float",
            filter: {
              attributes: {
                category: "size"
              }
            }
          }]
        }
      }
    }).buildAllPlatforms();

    describe(`flutter/class.dart`, () => {
      const output = fs.readFileSync(`${buildPath}style_dictionary.dart`, {encoding:`UTF-8`});

      it(`should match snapshot`, () => {
        expect(output).toMatchSnapshot();
      });

      describe(`with references`, () => {
        const output = fs.readFileSync(`${buildPath}style_dictionary_with_references.dart`, {encoding:`UTF-8`});

        it(`should match snapshot`, () => {
          expect(output).toMatchSnapshot();
        });

      });

      describe(`separate`, () => {
        const output = fs.readFileSync(`${buildPath}style_dictionary_color.dart`,{encoding:`UTF-8`});
        it(`should match snapshot`, () => {
          expect(output).toMatchSnapshot();
        });
      })
    });
  });
});

afterAll(() => {
  fs.emptyDirSync(buildPath);
});