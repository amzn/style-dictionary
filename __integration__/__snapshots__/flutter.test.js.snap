// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration flutter flutter/class.dart separate should match snapshot 1`] = `
"
//
// style_dictionary_color.dart
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT



import 'dart:ui';

class StyleDictionaryColor {
  StyleDictionaryColor._();

    static const backgroundDanger = Color(0xFFFFEAE9);
    static const backgroundDisabled = Color(0xFFDEE1E1);
    static const backgroundInfo = Color(0xFFE9F8FF);
    static const backgroundPrimary = Color(0xFFFFFFFF);
    static const backgroundSecondary = Color(0xFFF3F4F4);
    static const backgroundSuccess = Color(0xFFEBF9EB);
    static const backgroundTertiary = Color(0xFFDEE1E1);
    static const backgroundWarning = Color(0xFFFFEDE3);
    static const borderPrimary = Color(0xFFC8CCCC);
    static const brandPrimary = Color(0xFF0B8599);
    static const brandSecondary = Color(0xFF6F5ED3);
    static const coreAqua0 = Color(0xFFD9FCFB);
    static const coreAqua100 = Color(0xFFC5F9F9);
    static const coreAqua1000 = Color(0xFF083D4F);
    static const coreAqua1100 = Color(0xFF002838);
    static const coreAqua200 = Color(0xFFA5F2F2);
    static const coreAqua300 = Color(0xFF76E5E2);
    static const coreAqua400 = Color(0xFF33D6E2);
    static const coreAqua500 = Color(0xFF17B8CE);
    static const coreAqua600 = Color(0xFF0797AE);
    static const coreAqua700 = Color(0xFF0B8599);
    static const coreAqua800 = Color(0xFF0F6E84);
    static const coreAqua900 = Color(0xFF035E73);
    static const coreBlue0 = Color(0xFFE9F8FF);
    static const coreBlue100 = Color(0xFFDCF2FF);
    static const coreBlue1000 = Color(0xFF0A3960);
    static const coreBlue1100 = Color(0xFF002138);
    static const coreBlue200 = Color(0xFFC7E4F9);
    static const coreBlue300 = Color(0xFFA1D2F8);
    static const coreBlue400 = Color(0xFF56ADF5);
    static const coreBlue500 = Color(0xFF3896E3);
    static const coreBlue600 = Color(0xFF2B87D3);
    static const coreBlue700 = Color(0xFF2079C3);
    static const coreBlue800 = Color(0xFF116DAA);
    static const coreBlue900 = Color(0xFF0C5689);
    static const coreGreen0 = Color(0xFFEBF9EB);
    static const coreGreen100 = Color(0xFFD7F4D7);
    static const coreGreen1000 = Color(0xFF08422F);
    static const coreGreen1100 = Color(0xFF002B20);
    static const coreGreen200 = Color(0xFFC2F2BD);
    static const coreGreen300 = Color(0xFF98E58E);
    static const coreGreen400 = Color(0xFF75DD66);
    static const coreGreen500 = Color(0xFF59CB59);
    static const coreGreen600 = Color(0xFF2BB656);
    static const coreGreen700 = Color(0xFF0CA750);
    static const coreGreen800 = Color(0xFF008B46);
    static const coreGreen900 = Color(0xFF006B40);
    static const coreMagenta0 = Color(0xFFFEF0FF);
    static const coreMagenta100 = Color(0xFFF9E3FC);
    static const coreMagenta1000 = Color(0xFF451551);
    static const coreMagenta1100 = Color(0xFF29192D);
    static const coreMagenta200 = Color(0xFFF4C4F7);
    static const coreMagenta300 = Color(0xFFEDADF2);
    static const coreMagenta400 = Color(0xFFF282F5);
    static const coreMagenta500 = Color(0xFFDB61DB);
    static const coreMagenta600 = Color(0xFFC44EB9);
    static const coreMagenta700 = Color(0xFFAC44A8);
    static const coreMagenta800 = Color(0xFF8F3896);
    static const coreMagenta900 = Color(0xFF6C2277);
    static const coreNeutral0 = Color(0xFFFFFFFF);
    static const coreNeutral100 = Color(0xFFF3F4F4);
    static const coreNeutral1000 = Color(0xFF162020);
    static const coreNeutral1100 = Color(0xFF040404);
    static const coreNeutral200 = Color(0xFFDEE1E1);
    static const coreNeutral300 = Color(0xFFC8CCCC);
    static const coreNeutral400 = Color(0xFFB0B6B7);
    static const coreNeutral500 = Color(0xFF929A9B);
    static const coreNeutral600 = Color(0xFF6E797A);
    static const coreNeutral700 = Color(0xFF515E5F);
    static const coreNeutral800 = Color(0xFF364141);
    static const coreNeutral900 = Color(0xFF273333);
    static const coreOrange0 = Color(0xFFFFEDE3);
    static const coreOrange100 = Color(0xFFFCDCCC);
    static const coreOrange1000 = Color(0xFF601700);
    static const coreOrange1100 = Color(0xFF2D130E);
    static const coreOrange200 = Color(0xFFFFC6A4);
    static const coreOrange300 = Color(0xFFFFB180);
    static const coreOrange400 = Color(0xFFFF9C5D);
    static const coreOrange500 = Color(0xFFFC8943);
    static const coreOrange600 = Color(0xFFF57D33);
    static const coreOrange700 = Color(0xFFED7024);
    static const coreOrange800 = Color(0xFFCE5511);
    static const coreOrange900 = Color(0xFF962C0B);
    static const corePink0 = Color(0xFFFFE9F3);
    static const corePink100 = Color(0xFFFCDBEB);
    static const corePink1000 = Color(0xFF561231);
    static const corePink1100 = Color(0xFF2B1721);
    static const corePink200 = Color(0xFFFFB5D5);
    static const corePink300 = Color(0xFFFF95C1);
    static const corePink400 = Color(0xFFFF76AE);
    static const corePink500 = Color(0xFFEF588B);
    static const corePink600 = Color(0xFFE0447C);
    static const corePink700 = Color(0xFFCE3665);
    static const corePink800 = Color(0xFFB22F5B);
    static const corePink900 = Color(0xFF931847);
    static const corePurple0 = Color(0xFFF2F2F9);
    static const corePurple100 = Color(0xFFEAEAF9);
    static const corePurple1000 = Color(0xFF2D246B);
    static const corePurple1100 = Color(0xFF1D1D38);
    static const corePurple200 = Color(0xFFD8D7F9);
    static const corePurple300 = Color(0xFFC1C1F7);
    static const corePurple400 = Color(0xFFA193F2);
    static const corePurple500 = Color(0xFF9180F4);
    static const corePurple600 = Color(0xFF816FEA);
    static const corePurple700 = Color(0xFF6F5ED3);
    static const corePurple800 = Color(0xFF5E4EBA);
    static const corePurple900 = Color(0xFF483A9C);
    static const coreRed0 = Color(0xFFFFEAE9);
    static const coreRed100 = Color(0xFFFFD5D2);
    static const coreRed1000 = Color(0xFF6D1313);
    static const coreRed1100 = Color(0xFF2B1111);
    static const coreRed200 = Color(0xFFFFB8B1);
    static const coreRed300 = Color(0xFFFF9C8F);
    static const coreRed400 = Color(0xFFFF7F6E);
    static const coreRed500 = Color(0xFFF76054);
    static const coreRed600 = Color(0xFFED4C42);
    static const coreRed700 = Color(0xFFDB3E3E);
    static const coreRed800 = Color(0xFFC63434);
    static const coreRed900 = Color(0xFF992222);
    static const coreTeal0 = Color(0xFFE5F9F5);
    static const coreTeal100 = Color(0xFFCDF7EF);
    static const coreTeal1000 = Color(0xFF083F3F);
    static const coreTeal1100 = Color(0xFF002528);
    static const coreTeal200 = Color(0xFFB3F2E6);
    static const coreTeal300 = Color(0xFF7DEAD5);
    static const coreTeal400 = Color(0xFF24E0C5);
    static const coreTeal500 = Color(0xFF08C4B2);
    static const coreTeal600 = Color(0xFF00A99C);
    static const coreTeal700 = Color(0xFF0B968F);
    static const coreTeal800 = Color(0xFF067C7C);
    static const coreTeal900 = Color(0xFF026661);
    static const coreYellow0 = Color(0xFFFFF8E2);
    static const coreYellow100 = Color(0xFFFDEFCD);
    static const coreYellow1000 = Color(0xFF542A00);
    static const coreYellow1100 = Color(0xFF2D1A05);
    static const coreYellow200 = Color(0xFFFFE99A);
    static const coreYellow300 = Color(0xFFFFE16E);
    static const coreYellow400 = Color(0xFFFFD943);
    static const coreYellow500 = Color(0xFFFFCD1C);
    static const coreYellow600 = Color(0xFFFFBC00);
    static const coreYellow700 = Color(0xFFDD9903);
    static const coreYellow800 = Color(0xFFBA7506);
    static const coreYellow900 = Color(0xFF944C0C);
    static const fontDanger = Color(0xFF6D1313);
    static const fontInteractive = Color(0xFF0B8599);
    static const fontInteractiveActive = Color(0xFF6F5ED3);
    static const fontInteractiveDisabled = Color(0xFF364141);
    static const fontInteractiveHover = Color(0xFF0B8599);
    static const fontPrimary = Color(0xFF040404);
    static const fontSecondary = Color(0xFF273333);
    static const fontSuccess = Color(0xFF08422F);
    static const fontTertiary = Color(0xFF364141);
    static const fontWarning = Color(0xFF601700);
}"
`;

exports[`integration flutter flutter/class.dart should match snapshot 1`] = `
"
//
// style_dictionary.dart
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT



import 'dart:ui';

class StyleDictionary {
  StyleDictionary._();

    static const colorBackgroundDanger = Color(0xFFFFEAE9);
    static const colorBackgroundDisabled = Color(0xFFDEE1E1);
    static const colorBackgroundInfo = Color(0xFFE9F8FF);
    static const colorBackgroundPrimary = Color(0xFFFFFFFF);
    static const colorBackgroundSecondary = Color(0xFFF3F4F4);
    static const colorBackgroundSuccess = Color(0xFFEBF9EB);
    static const colorBackgroundTertiary = Color(0xFFDEE1E1);
    static const colorBackgroundWarning = Color(0xFFFFEDE3);
    static const colorBorderPrimary = Color(0xFFC8CCCC);
    static const colorBrandPrimary = Color(0xFF0B8599);
    static const colorBrandSecondary = Color(0xFF6F5ED3);
    static const colorCoreAqua0 = Color(0xFFD9FCFB);
    static const colorCoreAqua100 = Color(0xFFC5F9F9);
    static const colorCoreAqua1000 = Color(0xFF083D4F);
    static const colorCoreAqua1100 = Color(0xFF002838);
    static const colorCoreAqua200 = Color(0xFFA5F2F2);
    static const colorCoreAqua300 = Color(0xFF76E5E2);
    static const colorCoreAqua400 = Color(0xFF33D6E2);
    static const colorCoreAqua500 = Color(0xFF17B8CE);
    static const colorCoreAqua600 = Color(0xFF0797AE);
    static const colorCoreAqua700 = Color(0xFF0B8599);
    static const colorCoreAqua800 = Color(0xFF0F6E84);
    static const colorCoreAqua900 = Color(0xFF035E73);
    static const colorCoreBlue0 = Color(0xFFE9F8FF);
    static const colorCoreBlue100 = Color(0xFFDCF2FF);
    static const colorCoreBlue1000 = Color(0xFF0A3960);
    static const colorCoreBlue1100 = Color(0xFF002138);
    static const colorCoreBlue200 = Color(0xFFC7E4F9);
    static const colorCoreBlue300 = Color(0xFFA1D2F8);
    static const colorCoreBlue400 = Color(0xFF56ADF5);
    static const colorCoreBlue500 = Color(0xFF3896E3);
    static const colorCoreBlue600 = Color(0xFF2B87D3);
    static const colorCoreBlue700 = Color(0xFF2079C3);
    static const colorCoreBlue800 = Color(0xFF116DAA);
    static const colorCoreBlue900 = Color(0xFF0C5689);
    static const colorCoreGreen0 = Color(0xFFEBF9EB);
    static const colorCoreGreen100 = Color(0xFFD7F4D7);
    static const colorCoreGreen1000 = Color(0xFF08422F);
    static const colorCoreGreen1100 = Color(0xFF002B20);
    static const colorCoreGreen200 = Color(0xFFC2F2BD);
    static const colorCoreGreen300 = Color(0xFF98E58E);
    static const colorCoreGreen400 = Color(0xFF75DD66);
    static const colorCoreGreen500 = Color(0xFF59CB59);
    static const colorCoreGreen600 = Color(0xFF2BB656);
    static const colorCoreGreen700 = Color(0xFF0CA750);
    static const colorCoreGreen800 = Color(0xFF008B46);
    static const colorCoreGreen900 = Color(0xFF006B40);
    static const colorCoreMagenta0 = Color(0xFFFEF0FF);
    static const colorCoreMagenta100 = Color(0xFFF9E3FC);
    static const colorCoreMagenta1000 = Color(0xFF451551);
    static const colorCoreMagenta1100 = Color(0xFF29192D);
    static const colorCoreMagenta200 = Color(0xFFF4C4F7);
    static const colorCoreMagenta300 = Color(0xFFEDADF2);
    static const colorCoreMagenta400 = Color(0xFFF282F5);
    static const colorCoreMagenta500 = Color(0xFFDB61DB);
    static const colorCoreMagenta600 = Color(0xFFC44EB9);
    static const colorCoreMagenta700 = Color(0xFFAC44A8);
    static const colorCoreMagenta800 = Color(0xFF8F3896);
    static const colorCoreMagenta900 = Color(0xFF6C2277);
    static const colorCoreNeutral0 = Color(0xFFFFFFFF);
    static const colorCoreNeutral100 = Color(0xFFF3F4F4);
    static const colorCoreNeutral1000 = Color(0xFF162020);
    static const colorCoreNeutral1100 = Color(0xFF040404);
    static const colorCoreNeutral200 = Color(0xFFDEE1E1);
    static const colorCoreNeutral300 = Color(0xFFC8CCCC);
    static const colorCoreNeutral400 = Color(0xFFB0B6B7);
    static const colorCoreNeutral500 = Color(0xFF929A9B);
    static const colorCoreNeutral600 = Color(0xFF6E797A);
    static const colorCoreNeutral700 = Color(0xFF515E5F);
    static const colorCoreNeutral800 = Color(0xFF364141);
    static const colorCoreNeutral900 = Color(0xFF273333);
    static const colorCoreOrange0 = Color(0xFFFFEDE3);
    static const colorCoreOrange100 = Color(0xFFFCDCCC);
    static const colorCoreOrange1000 = Color(0xFF601700);
    static const colorCoreOrange1100 = Color(0xFF2D130E);
    static const colorCoreOrange200 = Color(0xFFFFC6A4);
    static const colorCoreOrange300 = Color(0xFFFFB180);
    static const colorCoreOrange400 = Color(0xFFFF9C5D);
    static const colorCoreOrange500 = Color(0xFFFC8943);
    static const colorCoreOrange600 = Color(0xFFF57D33);
    static const colorCoreOrange700 = Color(0xFFED7024);
    static const colorCoreOrange800 = Color(0xFFCE5511);
    static const colorCoreOrange900 = Color(0xFF962C0B);
    static const colorCorePink0 = Color(0xFFFFE9F3);
    static const colorCorePink100 = Color(0xFFFCDBEB);
    static const colorCorePink1000 = Color(0xFF561231);
    static const colorCorePink1100 = Color(0xFF2B1721);
    static const colorCorePink200 = Color(0xFFFFB5D5);
    static const colorCorePink300 = Color(0xFFFF95C1);
    static const colorCorePink400 = Color(0xFFFF76AE);
    static const colorCorePink500 = Color(0xFFEF588B);
    static const colorCorePink600 = Color(0xFFE0447C);
    static const colorCorePink700 = Color(0xFFCE3665);
    static const colorCorePink800 = Color(0xFFB22F5B);
    static const colorCorePink900 = Color(0xFF931847);
    static const colorCorePurple0 = Color(0xFFF2F2F9);
    static const colorCorePurple100 = Color(0xFFEAEAF9);
    static const colorCorePurple1000 = Color(0xFF2D246B);
    static const colorCorePurple1100 = Color(0xFF1D1D38);
    static const colorCorePurple200 = Color(0xFFD8D7F9);
    static const colorCorePurple300 = Color(0xFFC1C1F7);
    static const colorCorePurple400 = Color(0xFFA193F2);
    static const colorCorePurple500 = Color(0xFF9180F4);
    static const colorCorePurple600 = Color(0xFF816FEA);
    static const colorCorePurple700 = Color(0xFF6F5ED3);
    static const colorCorePurple800 = Color(0xFF5E4EBA);
    static const colorCorePurple900 = Color(0xFF483A9C);
    static const colorCoreRed0 = Color(0xFFFFEAE9);
    static const colorCoreRed100 = Color(0xFFFFD5D2);
    static const colorCoreRed1000 = Color(0xFF6D1313);
    static const colorCoreRed1100 = Color(0xFF2B1111);
    static const colorCoreRed200 = Color(0xFFFFB8B1);
    static const colorCoreRed300 = Color(0xFFFF9C8F);
    static const colorCoreRed400 = Color(0xFFFF7F6E);
    static const colorCoreRed500 = Color(0xFFF76054);
    static const colorCoreRed600 = Color(0xFFED4C42);
    static const colorCoreRed700 = Color(0xFFDB3E3E);
    static const colorCoreRed800 = Color(0xFFC63434);
    static const colorCoreRed900 = Color(0xFF992222);
    static const colorCoreTeal0 = Color(0xFFE5F9F5);
    static const colorCoreTeal100 = Color(0xFFCDF7EF);
    static const colorCoreTeal1000 = Color(0xFF083F3F);
    static const colorCoreTeal1100 = Color(0xFF002528);
    static const colorCoreTeal200 = Color(0xFFB3F2E6);
    static const colorCoreTeal300 = Color(0xFF7DEAD5);
    static const colorCoreTeal400 = Color(0xFF24E0C5);
    static const colorCoreTeal500 = Color(0xFF08C4B2);
    static const colorCoreTeal600 = Color(0xFF00A99C);
    static const colorCoreTeal700 = Color(0xFF0B968F);
    static const colorCoreTeal800 = Color(0xFF067C7C);
    static const colorCoreTeal900 = Color(0xFF026661);
    static const colorCoreYellow0 = Color(0xFFFFF8E2);
    static const colorCoreYellow100 = Color(0xFFFDEFCD);
    static const colorCoreYellow1000 = Color(0xFF542A00);
    static const colorCoreYellow1100 = Color(0xFF2D1A05);
    static const colorCoreYellow200 = Color(0xFFFFE99A);
    static const colorCoreYellow300 = Color(0xFFFFE16E);
    static const colorCoreYellow400 = Color(0xFFFFD943);
    static const colorCoreYellow500 = Color(0xFFFFCD1C);
    static const colorCoreYellow600 = Color(0xFFFFBC00);
    static const colorCoreYellow700 = Color(0xFFDD9903);
    static const colorCoreYellow800 = Color(0xFFBA7506);
    static const colorCoreYellow900 = Color(0xFF944C0C);
    static const colorFontDanger = Color(0xFF6D1313);
    static const colorFontInteractive = Color(0xFF0B8599);
    static const colorFontInteractiveActive = Color(0xFF6F5ED3);
    static const colorFontInteractiveDisabled = Color(0xFF364141);
    static const colorFontInteractiveHover = Color(0xFF0B8599);
    static const colorFontPrimary = Color(0xFF040404);
    static const colorFontSecondary = Color(0xFF273333);
    static const colorFontSuccess = Color(0xFF08422F);
    static const colorFontTertiary = Color(0xFF364141);
    static const colorFontWarning = Color(0xFF601700);
    static const sizeBorderRadiusLarge = 480.00;
    static const sizeFontLarge = 24.00;
    static const sizeFontMedium = 16.00;
    static const sizeFontSmall = 12.00;
    static const sizeFontXl = 36.00;
    static const sizePaddingLarge = 16.00;
    static const sizePaddingMedium = 16.00;
    static const sizePaddingSmall = 8.00;
    static const sizePaddingXl = 16.00;
}"
`;

exports[`integration flutter flutter/class.dart with references should match snapshot 1`] = `
"
//
// style_dictionary_with_references.dart
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT



import 'dart:ui';

class StyleDictionary {
  StyleDictionary._();

    static const sizePaddingXl = 16.00;
    static const sizePaddingLarge = 16.00;
    static const sizePaddingMedium = 16.00;
    static const sizePaddingSmall = 8.00;
    static const sizeFontXl = 36.00;
    static const sizeFontLarge = 24.00;
    static const sizeFontMedium = 16.00;
    static const sizeFontSmall = 12.00;
    static const sizeBorderRadiusLarge = 480.00;
    static const colorCoreYellow1100 = Color(0xFF2D1A05);
    static const colorCoreYellow1000 = Color(0xFF542A00);
    static const colorCoreYellow900 = Color(0xFF944C0C);
    static const colorCoreYellow800 = Color(0xFFBA7506);
    static const colorCoreYellow700 = Color(0xFFDD9903);
    static const colorCoreYellow600 = Color(0xFFFFBC00);
    static const colorCoreYellow500 = Color(0xFFFFCD1C);
    static const colorCoreYellow400 = Color(0xFFFFD943);
    static const colorCoreYellow300 = Color(0xFFFFE16E);
    static const colorCoreYellow200 = Color(0xFFFFE99A);
    static const colorCoreYellow100 = Color(0xFFFDEFCD);
    static const colorCoreYellow0 = Color(0xFFFFF8E2);
    static const colorCoreNeutral1100 = Color(0xFF040404);
    static const colorCoreNeutral1000 = Color(0xFF162020);
    static const colorCoreNeutral900 = Color(0xFF273333);
    static const colorCoreNeutral800 = Color(0xFF364141);
    static const colorCoreNeutral700 = Color(0xFF515E5F);
    static const colorCoreNeutral600 = Color(0xFF6E797A);
    static const colorCoreNeutral500 = Color(0xFF929A9B);
    static const colorCoreNeutral400 = Color(0xFFB0B6B7);
    static const colorCoreNeutral300 = Color(0xFFC8CCCC);
    static const colorCoreNeutral200 = Color(0xFFDEE1E1);
    static const colorCoreNeutral100 = Color(0xFFF3F4F4);
    static const colorCoreNeutral0 = Color(0xFFFFFFFF);
    static const colorCoreOrange1100 = Color(0xFF2D130E);
    static const colorCoreOrange1000 = Color(0xFF601700);
    static const colorCoreOrange900 = Color(0xFF962C0B);
    static const colorCoreOrange800 = Color(0xFFCE5511);
    static const colorCoreOrange700 = Color(0xFFED7024);
    static const colorCoreOrange600 = Color(0xFFF57D33);
    static const colorCoreOrange500 = Color(0xFFFC8943);
    static const colorCoreOrange400 = Color(0xFFFF9C5D);
    static const colorCoreOrange300 = Color(0xFFFFB180);
    static const colorCoreOrange200 = Color(0xFFFFC6A4);
    static const colorCoreOrange100 = Color(0xFFFCDCCC);
    static const colorCoreOrange0 = Color(0xFFFFEDE3);
    static const colorCoreRed1100 = Color(0xFF2B1111);
    static const colorCoreRed1000 = Color(0xFF6D1313);
    static const colorCoreRed900 = Color(0xFF992222);
    static const colorCoreRed800 = Color(0xFFC63434);
    static const colorCoreRed700 = Color(0xFFDB3E3E);
    static const colorCoreRed600 = Color(0xFFED4C42);
    static const colorCoreRed500 = Color(0xFFF76054);
    static const colorCoreRed400 = Color(0xFFFF7F6E);
    static const colorCoreRed300 = Color(0xFFFF9C8F);
    static const colorCoreRed200 = Color(0xFFFFB8B1);
    static const colorCoreRed100 = Color(0xFFFFD5D2);
    static const colorCoreRed0 = Color(0xFFFFEAE9);
    static const colorCorePink1100 = Color(0xFF2B1721);
    static const colorCorePink1000 = Color(0xFF561231);
    static const colorCorePink900 = Color(0xFF931847);
    static const colorCorePink800 = Color(0xFFB22F5B);
    static const colorCorePink700 = Color(0xFFCE3665);
    static const colorCorePink600 = Color(0xFFE0447C);
    static const colorCorePink500 = Color(0xFFEF588B);
    static const colorCorePink400 = Color(0xFFFF76AE);
    static const colorCorePink300 = Color(0xFFFF95C1);
    static const colorCorePink200 = Color(0xFFFFB5D5);
    static const colorCorePink100 = Color(0xFFFCDBEB);
    static const colorCorePink0 = Color(0xFFFFE9F3);
    static const colorCoreMagenta1100 = Color(0xFF29192D);
    static const colorCoreMagenta1000 = Color(0xFF451551);
    static const colorCoreMagenta900 = Color(0xFF6C2277);
    static const colorCoreMagenta800 = Color(0xFF8F3896);
    static const colorCoreMagenta700 = Color(0xFFAC44A8);
    static const colorCoreMagenta600 = Color(0xFFC44EB9);
    static const colorCoreMagenta500 = Color(0xFFDB61DB);
    static const colorCoreMagenta400 = Color(0xFFF282F5);
    static const colorCoreMagenta300 = Color(0xFFEDADF2);
    static const colorCoreMagenta200 = Color(0xFFF4C4F7);
    static const colorCoreMagenta100 = Color(0xFFF9E3FC);
    static const colorCoreMagenta0 = Color(0xFFFEF0FF);
    static const colorCorePurple1100 = Color(0xFF1D1D38);
    static const colorCorePurple1000 = Color(0xFF2D246B);
    static const colorCorePurple900 = Color(0xFF483A9C);
    static const colorCorePurple800 = Color(0xFF5E4EBA);
    static const colorCorePurple700 = Color(0xFF6F5ED3);
    static const colorCorePurple600 = Color(0xFF816FEA);
    static const colorCorePurple500 = Color(0xFF9180F4);
    static const colorCorePurple400 = Color(0xFFA193F2);
    static const colorCorePurple300 = Color(0xFFC1C1F7);
    static const colorCorePurple200 = Color(0xFFD8D7F9);
    static const colorCorePurple100 = Color(0xFFEAEAF9);
    static const colorCorePurple0 = Color(0xFFF2F2F9);
    static const colorCoreBlue1100 = Color(0xFF002138);
    static const colorCoreBlue1000 = Color(0xFF0A3960);
    static const colorCoreBlue900 = Color(0xFF0C5689);
    static const colorCoreBlue800 = Color(0xFF116DAA);
    static const colorCoreBlue700 = Color(0xFF2079C3);
    static const colorCoreBlue600 = Color(0xFF2B87D3);
    static const colorCoreBlue500 = Color(0xFF3896E3);
    static const colorCoreBlue400 = Color(0xFF56ADF5);
    static const colorCoreBlue300 = Color(0xFFA1D2F8);
    static const colorCoreBlue200 = Color(0xFFC7E4F9);
    static const colorCoreBlue100 = Color(0xFFDCF2FF);
    static const colorCoreBlue0 = Color(0xFFE9F8FF);
    static const colorCoreAqua1100 = Color(0xFF002838);
    static const colorCoreAqua1000 = Color(0xFF083D4F);
    static const colorCoreAqua900 = Color(0xFF035E73);
    static const colorCoreAqua800 = Color(0xFF0F6E84);
    static const colorCoreAqua700 = Color(0xFF0B8599);
    static const colorCoreAqua600 = Color(0xFF0797AE);
    static const colorCoreAqua500 = Color(0xFF17B8CE);
    static const colorCoreAqua400 = Color(0xFF33D6E2);
    static const colorCoreAqua300 = Color(0xFF76E5E2);
    static const colorCoreAqua200 = Color(0xFFA5F2F2);
    static const colorCoreAqua100 = Color(0xFFC5F9F9);
    static const colorCoreAqua0 = Color(0xFFD9FCFB);
    static const colorCoreTeal1100 = Color(0xFF002528);
    static const colorCoreTeal1000 = Color(0xFF083F3F);
    static const colorCoreTeal900 = Color(0xFF026661);
    static const colorCoreTeal800 = Color(0xFF067C7C);
    static const colorCoreTeal700 = Color(0xFF0B968F);
    static const colorCoreTeal600 = Color(0xFF00A99C);
    static const colorCoreTeal500 = Color(0xFF08C4B2);
    static const colorCoreTeal400 = Color(0xFF24E0C5);
    static const colorCoreTeal300 = Color(0xFF7DEAD5);
    static const colorCoreTeal200 = Color(0xFFB3F2E6);
    static const colorCoreTeal100 = Color(0xFFCDF7EF);
    static const colorCoreTeal0 = Color(0xFFE5F9F5);
    static const colorCoreGreen1100 = Color(0xFF002B20);
    static const colorCoreGreen1000 = Color(0xFF08422F);
    static const colorCoreGreen900 = Color(0xFF006B40);
    static const colorCoreGreen800 = Color(0xFF008B46);
    static const colorCoreGreen700 = Color(0xFF0CA750);
    static const colorCoreGreen600 = Color(0xFF2BB656);
    static const colorCoreGreen500 = Color(0xFF59CB59);
    static const colorCoreGreen400 = Color(0xFF75DD66);
    static const colorCoreGreen300 = Color(0xFF98E58E);
    static const colorCoreGreen200 = Color(0xFFC2F2BD);
    static const colorCoreGreen100 = Color(0xFFD7F4D7);
    static const colorCoreGreen0 = Color(0xFFEBF9EB);
    static const colorFontSuccess = colorCoreGreen1000;
    static const colorFontWarning = colorCoreOrange1000;
    static const colorFontDanger = colorCoreRed1000;
    static const colorFontTertiary = colorCoreNeutral800;
    static const colorFontSecondary = colorCoreNeutral900;
    static const colorFontPrimary = colorCoreNeutral1100;
    static const colorBrandSecondary = colorCorePurple700;
    static const colorBrandPrimary = colorCoreAqua700;
    static const colorBorderPrimary = colorCoreNeutral300;
    static const colorBackgroundInfo = colorCoreBlue0;
    static const colorBackgroundSuccess = colorCoreGreen0;
    static const colorBackgroundWarning = colorCoreOrange0;
    static const colorBackgroundDanger = colorCoreRed0;
    static const colorBackgroundTertiary = colorCoreNeutral200;
    static const colorBackgroundSecondary = colorCoreNeutral100;
    static const colorBackgroundPrimary = colorCoreNeutral0;
    static const colorFontInteractiveDisabled = colorFontTertiary;
    static const colorFontInteractiveActive = colorBrandSecondary;
    static const colorFontInteractiveHover = colorBrandPrimary;
    static const colorFontInteractive = colorBrandPrimary;
    static const colorBackgroundDisabled = colorBackgroundTertiary;
}"
`;
